kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/V3ResourcesModule.java

```diff
@@ -87,4 +111,50 @@ public void dispose(ExecutorService executorService) {
       }
     }
   }
+
+  private static final class StreamingResponseIdleTimeExecutorFactory
+      implements Factory<ExecutorService> {
+
+    @Override
+    public ExecutorService provide() {
+      return Executors.newCachedThreadPool(
+          new ThreadFactoryBuilder()
+              .setNameFormat("kafka-rest-streaming-response-idle-time-thread-%d")
+              .build());
+    }
+
+    @Override
+    public void dispose(ExecutorService executorService) {
+      executorService.shutdown();
+      try {
+        if (!executorService.awaitTermination(60, TimeUnit.SECONDS)) {
+          executorService.shutdownNow();
+        }
+      } catch (InterruptedException e) {
+        executorService.shutdownNow();
+      }
+    }
+  }
+
+  private static final class StreamingResponseIdleTimeLimiterFactory
+      implements Factory<StreamingResponseIdleTimeLimiter> {
+    private final ExecutorService executor;
+    private final Duration maxIdleTime;
+
+    @Inject
+    StreamingResponseIdleTimeLimiterFactory(
+        @StreamingResponseIdleTimeExecutor ExecutorService executor,
```

Comment:
Because we want to close it at server shutdown. One can get about doing the same by keeping a reference to the executor in the `StreamingResponseIdleTimeLimiter` class, but I find it a kind of anti-pattern to have aa owning-reference to something that you don't use for anything except closing. It is, though, an alternative way to do it, which would require about the same amount the code.

Reply:
I disagree, and would have insisted more on this, have this not been a hotfix. I also think that the alternative would have been more readable, or at least easier to understand - see my rant about our usage of HK2.
And wouldn't that alternative just manage the lifecycle of that executor in the idle limiter factory? I.e. create the executor when the factory is created, dispose of it when the provided singleton limiter is disposed?